src/win/pipe.c

```diff
@@ -766,32 +887,53 @@ int uv_pipe_bind2(uv_pipe_t* handle,
     req->next_pending = NULL;
   }
 
-  /* TODO(bnoordhuis) Add converters that take a |length| parameter. */
-  err = uv__convert_utf8_to_utf16(name_copy, &handle->name);
-  uv__free(name_copy);
-  name_copy = NULL;
+  if (!use_uds_pipe) {
+    /* TODO(bnoordhuis) Add converters that take a |length| parameter. */
+    err = uv__convert_utf8_to_utf16(name_copy, &handle->name);
+    uv__free(name_copy);
+    name_copy = NULL;
+  } else {
+    /* Use unix domain socket we save the original path name copy. */
+    handle->pathname = name_copy;
+    name_copy = NULL;
+  }
 
   if (err) {
     goto error;
   }
 
-  /*
-   * Attempt to create the first pipe with FILE_FLAG_FIRST_PIPE_INSTANCE.
-   * If this fails then there's already a pipe server for the given pipe name.
-   */
-  if (!pipe_alloc_accept(loop,
-                         handle,
-                         &handle->pipe.serv.accept_reqs[0],
-                         TRUE)) {
-    err = GetLastError();
-    if (err == ERROR_ACCESS_DENIED) {
-      err = UV_EADDRINUSE;
-    } else if (err == ERROR_PATH_NOT_FOUND || err == ERROR_INVALID_NAME) {
-      err = UV_EACCES;
-    } else {
-      err = uv_translate_sys_error(err);
+  if (use_uds_pipe) {
+    /*
+     * Prefix not match named pipe, use unix domain socket.
+     */
+#if defined(UV__ENABLE_WIN_UDS_PIPE)
```

Comment:
If this is not defined, the if will never be true, so it will never be here.

But define here inside can help prevent affecting the `else` statement.

Reply:
Refined.